spring-integration-file/src/test/java/org/springframework/integration/file/remote/session/ContextHolderRequestHandlerAdviceTests.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.remote.session;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.MessageChannels;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.stereotype.Component;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * @author Adel Haidar
+ *
+ * @since 6.1
+ *
+ */
+@SpringJUnitConfig
+public class ContextHolderRequestHandlerAdviceTests {
+
+	@Autowired
+	TestSessionFactory foo;
+
+	@Autowired
+	TestSessionFactory bar;
+
+	@Autowired
+	Config config;
+
+	@Test
+	public void testFlow() {
+		final Map<String, Object> headers = Map.of("test-key", "foo");
+		config.in().send(new GenericMessage<>("any-payload", headers));
+		Message<?> received = config.out().receive(0);
+		assertThat(received).isNotNull();
+		assertThat(Config.testSessionDuringHandling).isNotNull().isEqualTo(foo.getSession());
+		assertThat(config.dsf().getSession()).isEqualTo(bar.getSession());
```

Comment:
I think this test is still meaningless and configuration is too complicated.

I believe we simple could test against some `ThreadLocal` used in the `@ServiceActivator` method with such an advice.
And then in the test we ensure that the value from that `ThreadLocal` has been accessed from that method.
Plus in the end verify that `ThreadLocal` is cleared according to the `ContextHolderRequestHandlerAdvice` expectations.

Then `DelegatingSessionFactoryTests` might be modified for this `ContextHolderRequestHandlerAdvice` instead of those `<int:service-activator>`.
And perhaps the whole XML config could be removed altogether since the logic for context holder is supported via `ContextHolderRequestHandlerAdvice` now.

Reply:
I am sorry, I haven't got the chance to properly work on this yet. That's why I just pushed the easy stuff and left this for later. I honestly didn't quite understand what you mean. But I will try anyway.

Reply:
I don't understand what you meant by the following:

> I believe we simple could test against some ThreadLocal used in the @ServiceActivator method with such an advice.